scipy/integrate/_quadpack_py.py

```diff
@@ -1168,7 +1181,19 @@ def nquad(func, ranges, args=None, opts=None, full_output=False):
         opts = [_OptFunc(opts)] * depth
     else:
         opts = [opt if callable(opt) else _OptFunc(opt) for opt in opts]
-    return _NQuad(func, ranges, opts, full_output).integrate(*args)
+    res = _NQuad(func, ranges, opts, True).integrate(*args)
+
+    if full_output is not None:
+        msg = ("Use of parameter `full_output` is deprecated. Please leave "
+               "`full_output` unspecified; `neval` can now be accessed as an "
+               "attribute of the object returned by `scipy.integrate.nquad`.")
+        warnings.warn(msg, DeprecationWarning, stacklevel=2)
+
+    if full_output:
+        return res
```

Comment:
Should this not be something along the lines of
```suggestion
        return return QuadratureResult(integral=res[0], abserr=res[1], neval=None)
```
otherwise the function is returning a tuple opposed to the results object as claimed by the documentation?